pkg/serverless/invocationlifecycle/trace_test.go

```diff
@@ -14,20 +14,22 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+var testString = `a5a{"resource":"/users/create","path":"/users/create","httpMethod":"GET","headers":{"Accept":"*/*","Accept-Encoding":"gzip","x-datadog-parent-id":"1480558859903409531","x-datadog-sampling-priority":"1","x-datadog-trace-id":"5736943178450432258"}}0`
+
 func TestStartExecutionSpan(t *testing.T) {
```

Comment:
It's great that you added test coverage for the new case, however I think we should also keep the test coverage for what was being tested before (e.g. no payload) and also add the case of a payload without valid trace headers. Also, what about a non-API Gateway payload? We don't need to handle those correctly yet, but we should verify that the code behaves as expected.